doc/architecture.md

```diff
@@ -0,0 +1,136 @@
+Architecture
+============
+
+This section describes how a Management Pack works, and how it interacts with the VMware 
+Aria Operations Collector.
+
+## Management Pack
+A management pack consists of content, metadata, and an adapter.
+
+* Content includes Dashboards, Reports, Icons, Symptoms, Alerts, Traversals, others.
+* Metadata includes the describe.xml file that describes an adapter, and the 
+  `manifest.txt` file that describes the management pack as a whole.
+* The Adapter is the code that performs the collection and sends the data to VMware Aria 
+  Operations.
+
+### Adapter Container Image
+
+The adapter container image consists of two parts:
+* A REST server specified by [Collector Framework 2](../vmware_aria_operations_integration_sdk/api/vmware-aria-operations-collector-fwk2.json).
+* The adapter code that performs the collection.
+
+In an adapter created using the SDK, the REST server is provided inside a base container
+image that can built upon. In the adapter's dockerfile, this is specified in the 'FROM' 
+directive on the first line. This default server implements all the required endpoints, 
+provides error handling, and adds some additional context to the user code that is not 
+present in the original REST requests. The server also adds an additional 
+[definition endpoint](../vmware_aria_operations_integration_sdk/api/integration-sdk-definition-endpoint.json)
+which allows the collection code to describe itself, rather than requiring a 
+`describe.xml` file to be manually created, and some features that improve the debugging
+experience.
+
+The server determines how to call the user code by reading a file called `commands.cfg` 
+that is present in the Project Directory. This file provides a mapping from each 
+endpoint served by the REST server to an executable command that will run as a separate 
+process inside the container. Importantly, the server appends two extra arguments to 
+each process call: An input and output named pipe. These are used for interprocess 
+communication.
+
+Because commands are run as a subprocess, the server is able to detect and handle many
+common errors (such as the process crashing without writing a result) without affecting
+the operation of the server.
+
+Importantly, this model enforces that adapters remain stateless. Because each call 
```

Comment:
To help reduce complexity, I think this document should focus on the "what" and not go too much into the "why". I think the "why" is important here, but maybe we can find a way to sequester or move it to another page.

Reply:
Yeah, I agree. I've removed it for now, but will try to find a better place for it elsewhere